wgsl/index.bs

```diff
@@ -3959,6 +3959,278 @@ TODO: deduplicate these tables
+
+```
+<result> textureSample(<texture>, <sampler>, <coordinates>, bias: <f32> [, <named-arguments>])
+```
+
+Applies a bias to the mip level before sampling.
+
+The bias value must be between `-16.0` and `15.99`.
+
+<div class='example' heading="Sample with bias">
+  <xmp>
+    color = textureSample(tex2D, s, coords, bias: 1.0);
+  </xmp>
+</div>
+
+#### Sample with explicit LOD #### {#texture-sample-with-lod}
+
+```
+<result> textureSample(<texture>, <sampler>, <coordinates>, lod: <f32> [, <named-arguments>])
+```
+
+Specifies the explicit mip level to sample.
+Fractional values may interpolate between two levels.
+
+<div class='example' heading="Sample with explicit LOD">
+  <xmp>
+    color = textureSample(tex2D, s, coords, lod: 1.0);
+  </xmp>
+</div>
+
+#### Sample with explicit gradients #### {#texture-sample-with-grad}
+
+```
+<result> textureSample(<texture>, <sampler>, <coordinates>, grad: <gradients> [, <named-arguments>])
+```
+
+Uses the gradient specified by the gradient vectors in the two-dimensional array
+`<gradients>` to compute the sampling locations.
+
+`<gradients>` is of type `array<T, 2>`, where the first element holds the `ddx`
+derivative and the second holds the `ddy` derivitive.
+The type `T` is dependent on the type of `<texture>`, and is either a
+`vec2<f32>` or `vec3<f32>`:
+
+<table class='data'>
+  <thead>
+    <tr><td>`<texture>` type<td>`T` type
+  </thead>
+  <tr><td> `texture_sampled_2d<type>`         <td> `vec2<f32>`
+  <tr><td> `texture_sampled_2d_array<type>`   <td> `vec2<f32>`
+  <tr><td> `texture_sampled_3d<type>`         <td> `vec3<f32>`
+  <tr><td> `texture_sampled_cube<type>`       <td> `vec3<f32>`
+  <tr><td> `texture_sampled_cube_array<type>` <td> `vec3<f32>`
+  <tr><td> `texture_ro_2d<type>`              <td> `vec2<f32>`
+  <tr><td> `texture_ro_2d_array<type>`        <td> `vec2<f32>`
+  <tr><td> `texture_ro_3d<type>`              <td> `vec3<f32>`
+  <tr><td> `texture_depth_2d`                 <td> `vec2<f32>`
+  <tr><td> `texture_depth_2d_array`           <td> `vec2<f32>`
+  <tr><td> `texture_depth_cube`               <td> `vec3<f32>`
+  <tr><td> `texture_depth_cube_array`         <td> `vec3<f32>`
+</table>
+
+<div class='example' heading="Sample with explicit gradients">
+  <xmp>
+    var gradients : array<vec2<f32>, 2> = array<vec2<f32>, 2>(calcDDX(), calcDDY());
+    color = textureSample(tex2D, s, coords, grad: gradients);
+  </xmp>
+</div>
+
+#### Named Argument Compatibility #### {#texture-sample-named-arg-compatibility}
+
+Not all additional named arguments are compatible with every texture type, nor with each other.
+Consult the following tables for compatibility:
+
+<table class='data'>
+  <caption>Compatibility with texture type</caption>
+  <thead>
+    <tr><td>`<texture>` type                  <td>`offset`<td>`bias`  <td>`lod`   <td>`grad`
+  </thead>
+  <tr><td> `texture_sampled_1d<type>`         <td>&#10003;<td>        <td>        <td>
```

Comment:
I wonder if we could use named special symbols instead of the codes

Reply:
Maybe - I didn't see anything in the bikeshed docs when I did a quick scan for this. Anyone know of any better way?

Reply:
I used "&checkmark" in https://github.com/gpuweb/gpuweb/blob/9cd16d59c328a35233e8d04a4ee24de72b9fcbea/spec/index.bs#L7152

Reply:
That works. Thanks! Done.